fdbserver/storageserver.actor.cpp

```diff
@@ -3476,14 +3511,19 @@ ACTOR Future<Void> replaceInterface( StorageServer* self, StorageServerInterface
 	return Void();
 }
 
-ACTOR Future<Void> storageServer( IKeyValueStore* persistentData, StorageServerInterface ssi, Reference<AsyncVar<ServerDBInfo>> db, std::string folder, Promise<Void> recovered )
+ACTOR Future<Void> storageServer( IKeyValueStore* persistentData, StorageServerInterface ssi, Reference<AsyncVar<ServerDBInfo>> db, std::string folder, Promise<Void> recovered)
 {
 	state StorageServer self(persistentData, db, ssi);
 	self.folder = folder;
 	self.sk = serverKeysPrefixFor( self.thisServerID ).withPrefix(systemKeys.begin);  // FFFF/serverKeys/[this server]/
 	try {
 		state double start = now();
 		TraceEvent("StorageServerRebootStart", self.thisServerID);
+		bool dispose = wait(memoryStoreRecover (persistentData, self.cx, self.thisServerID) );
```

Comment:
I think what you want is to run `memoryStoreRecover` until it returns `false` and then run it concurrently with `restoreDurableState` below?